src/builtins/core/duration/tests.rs

```diff
@@ -190,6 +190,12 @@ fn duration_from_str() {
     assert_eq!(duration.nanoseconds(), 940);
 }
 
+#[test]
+fn duration_large_invalid() {
```

Comment:
I follow up in the issue.

Can you add all 3 of the different maximum duration values to this test?